### PR TITLE
Update encrypted store from throwing if decryption operation fails

### PR DESCRIPTION
**Pull request by varunpuranik**
If the key used to decrypt changes in the iotedged, the decryption operation fails. 
If the EdgeHub has its storage volume mounted into the container, this currently causes the EdgeHub to crash. 
This PR fixes this by updating the EncryptionProvider to treat a decryption failure as "value not found".
Also changed the retry/backoff strategy for the Workload client, to avoid the decryption failure to be retried too many times and for too long, causing the EdgeHub to back off.

The ideal fix for this would be to return a different status from the iotedged if decryption fails instead of 500, which can be handled differently. But looks like that is tricky to do at the moment, because of the way the hsm lib is implemented.

### CODE REVIEW

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClientVersioned.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.Azure.Devices.Edge.Util.Edged
     abstract class WorkloadClientVersioned
     {
         static readonly RetryStrategy TransientRetryStrategy =
-            new ExponentialBackoff(retryCount: 3, minBackoff: TimeSpan.FromSeconds(2), maxBackoff: TimeSpan.FromSeconds(30), deltaBackoff: TimeSpan.FromSeconds(3));
+            new ExponentialBackoff(retryCount: 2, minBackoff: TimeSpan.FromSeconds(1), maxBackoff: TimeSpan.FromSeconds(3), deltaBackoff: TimeSpan.FromSeconds(2));
```
Comment (myagley) [inline edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClientVersioned.cs]: Why the change here?

File: edge-util/src/Microsoft.Azure.Devices.Edge.Storage/EncryptedStore.cs [hunk 2]
```diff
@@ -9,14 +9,15 @@ namespace Microsoft.Azure.Devices.Edge.Storage
     public class EncryptedStore<TK, TV> : IKeyValueStore<TK, TV>
```
Comment (myagley) [inline edge-util/src/Microsoft.Azure.Devices.Edge.Storage/EncryptedStore.cs]: This change will work for encrypting twin as well?

Comment (varunpuranik) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Storage/EncryptedStore.cs]: Yes, this and the change in UpdatableEncryptedStore

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClientVersioned.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClientVersioned.cs]: This is because, currently it is not possible to distinguish between iotedged operations failing because it cannot decrypt vs some other issue. So all operations are retried. 
This is to reduce the time taken for retrying all the decrypt operations, to prevent EdgeHub initialization from timing out.
